Assets/Scripts/EnemyBehaviour/EnemyBehaviour.cs

```diff
@@ -5,10 +5,21 @@
 public class EnemyBehaviour : EntityBehaviour
 {
     public int maxHealthIncreasePerLoop = 2;
+    [SerializeField] private enemyType thisEnemyType;
     protected override void Awake()
     {
         //Increase maxhealth by 2 per loop
         maxHealth += (CarriageManager.loopsCompleted * maxHealthIncreasePerLoop);
         base.Awake();
     }
+    protected virtual void Start()
+    {
+        thisCombatManager.UpdateCountOfEachEnemyType(1, (int)thisEnemyType);
+    }
+    public override void Death()
+    {
+        //Tell combat manager to remove active count
+        thisCombatManager.UpdateCountOfEachEnemyType(-1, (int)thisEnemyType);
```

Comment:
could you explain why you're casting enums back into integers? why not just use it as the enum type?

Reply:
The short answer is, previous code prior to starting using enum had been with int.
So I basically just did as such, to avoid having to tinker with old code and potentially mess anything.